### PR TITLE
clientv3: fix lease "freezing" on unhealthy cluster

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: clientv3/integration/lease_test.go [hunk 1]
```diff
@@ -574,3 +576,54 @@ func TestLeaseKeepAliveLoopExit(t *testing.T) {
...
+				}
+				t.Fatal(err)
+			}()
+		}
+	}
+
+	mkReqs(1)
+	clus.Members[1].Stop(t)
+	mkReqs(2)
+	time.Sleep(time.Second)
```
Comment (gyuho) [inline clientv3/integration/lease_test.go]: Q. Why do we wait here?

Comment (heyitsanthony) [inline clientv3/integration/lease_test.go]: otherwise the leader resumes too fast; couldn't trigger the failure without it

**Comment by fanminshi** [comment]
what was the original issue that this pr is trying to fix?

**Comment by heyitsanthony** [comment]
@fanminshi 
1. error wasn't coded as `Unavailable` so it was causing the lease keep alive loop to terminate.
2. if `Grant` received a non-halting error, it would try to reset the keepalive loop stream, which would cancel outstanding keepalives.

**Comment by xiang90** [comment]
lgtm

probably worth to add the explanation https://github.com/coreos/etcd/pull/7023#issuecomment-267657040 to the commit message.

**Pull request merged**